launch_ros/launch_ros/actions/node.py

```diff
@@ -365,7 +365,19 @@ def execute(self, context: LaunchContext) -> Optional[List[Action]]:
                     '{}:={}'.format(remapping_from, remapping_to)
                 )
         context.extend_locals({'ros_specific_arguments': ros_specific_arguments})
-        return super().execute(context)
+        ret = super().execute(context)
+
+        self.__logger = launch.logging.get_logger(self.name)
+        try:
+            unique_node_names = context.locals.unique_ros_node_names
+        except AttributeError:
+            unique_node_names = set()
+        if self.node_name in unique_node_names:
+            self.__logger.warning('{} is not a unique node name'.format(self.node_name))
```

Comment:
This wording may be confusing - at the risk of being too verbose, I think we should be as explicit as possible, perhaps "there is more than 1 node with the exact name '{}' being created within this launch context"

Reply:
This should no longer be applicable after https://github.com/ros2/launch_ros/pull/127#discussion_r385871278 is resolved.